### ISSUE TITLE
Custom metrics that depends on image_path?

### ISSUE THREAD

**Issue by Anton-Cherepkov** [Deci-AI/super-gradients#2009]
### 💡 Your Question

Hello, community

**Problem description**:
I want to create a custom class `PrecisionRecallROIMetrics(DetectionMetrics)`.

The class `PrecisionRecallMetrics` will contain logic for calculating detection metrics (e.g. precision, recall) for some Region of Interest (arbitrary region of image defined by polygon). The Region of Interest will be unique for each image in the dataset and can be somehow retrieved by the path to the image.

My current problem with implementing such a class is the following:
- the `update` method of `DetectionMetrics` [has fixed inputs](https://github.com/Deci-AI/super-gradients/blob/217353a3a2fc919d2330d9c74f78e40d96d72b35/src/super_gradients/training/metrics/detection_metrics.py#L166) which do not include the img_paths:
```
def update(
  self,
  preds,
  target: torch.Tensor,
  device: str,
  inputs: torch.tensor,
  crowd_targets: Optional[torch.Tensor] = None) -> None:
```

**Question:**
Is there any easy way to populate the `img_path` to the `DetectionMetrics` object?
